tests/test_scheduler.py

```diff
@@ -1,68 +1,77 @@
+
+from aiida import orm
 from aiida.schedulers.datastructures import CodeRunMode, JobTemplate
 import pytest
 
 from aiida_firecrest.scheduler import FirecrestScheduler
-from aiida_firecrest.transport import FirecrestTransport
-from aiida_firecrest.utils_test import FirecrestConfig
-
-
-@pytest.fixture(name="transport")
-def _transport(firecrest_server: FirecrestConfig):
-    transport = FirecrestTransport(
-        url=firecrest_server.url,
-        token_uri=firecrest_server.token_uri,
-        client_id=firecrest_server.client_id,
-        client_secret=firecrest_server.client_secret,
-        client_machine=firecrest_server.machine,
-        small_file_size_mb=firecrest_server.small_file_size_mb,
-    )
-    transport.chdir(firecrest_server.scratch_path)
-    yield transport
-
-
-def test_get_jobs_empty(transport: FirecrestTransport):
-    scheduler = FirecrestScheduler()
-    scheduler.set_transport(transport)
-
-    with pytest.raises(SchedulerError, match="Failed to retrieve job"):
-        scheduler.get_jobs(["unknown"])
-
-    assert isinstance(scheduler.get_jobs(), list)
+from conftest import Values
 
 
-def test_submit_job(transport: FirecrestTransport):
+@pytest.mark.usefixtures("aiida_profile_clean")
+def test_submit_job(firecrest_computer: orm.Computer, tmp_path: Path):
+    transport = firecrest_computer.get_transport()
     scheduler = FirecrestScheduler()
     scheduler.set_transport(transport)
 
-    with pytest.raises(SchedulerError, match="invalid path"):
+    with pytest.raises(FileNotFoundError):
         scheduler.submit_job(transport.getcwd(), "unknown.sh")
 
-    # create a job script in a folder
-    transport.mkdir("test_submission")
-    transport.chdir("test_submission")
-    transport.write_binary("job.sh", b"#!/bin/bash\n\necho 'hello world'")
+    _script = Path(tmp_path / "job.sh")
+    _script.write_text("#!/bin/bash\n\necho 'hello world'")
 
-    job_id = scheduler.submit_job(transport.getcwd(), "job.sh")
+    job_id = scheduler.submit_job(transport.getcwd(), _script)
+    # this is how aiida expects the job_id to be returned
     assert isinstance(job_id, str)
 
 
-def test_write_script_minimal(file_regression):
+@pytest.mark.usefixtures("aiida_profile_clean")
+def test_get_jobs(firecrest_computer: orm.Computer):
+    transport = firecrest_computer.get_transport()
     scheduler = FirecrestScheduler()
-    template = JobTemplate(
-        {
-            "job_resource": scheduler.create_job_resource(
-                num_machines=1, num_mpiprocs_per_machine=1
-            ),
-            "codes_info": [],
-            "codes_run_mode": CodeRunMode.SERIAL,
-        }
-    )
-    file_regression.check(scheduler.get_submit_script(template).rstrip() + "\n")
-
+    scheduler.set_transport(transport)
 
-def test_write_script_full(file_regression):
+    # test pagaination
+    scheduler._DEFAULT_PAGE_SIZE = 2
+    Values._DEFAULT_PAGE_SIZE = 2
+
+    joblist = [random.randint(10000, 99999) for i in range(5)]
```

Comment:
Should the test also work if some of the job idi don't exist? If yes, maybe just hardcode the numbers (easier for debugging)?